rcl/src/rcl/wait.c

```diff
@@ -77,6 +78,7 @@ __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
     assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
   }
   if (wait_set->impl) {
+    assert(rmw_destroy_waitset(wait_set->impl->rmw_waitset) == RMW_RET_OK);
```

Comment:
This needs to be guarded by a `if (wait_set->impl->rmw_waitset) {...}` because it is possible for allocation of `wait_set->impl` to be successful, but `wait_set->impl->rmw_waitset` is still NULL or does not point to allocated memory.